firstvoices/backend/serializers/utils.py

```diff
@@ -1,5 +1,13 @@
+from django.http import Http404
+
+from backend.models.sites import Site
+
+
 def get_site_from_context(serializer):
     if "site" in serializer.context:
         return serializer.context["site"]
-    else:
-        return None
+    elif "site_slug" in serializer.context:
+        site = Site.objects.filter(slug=serializer.context["site_slug"])
+        if len(site) == 0:
+            return Http404
```

Comment:
It's great to check whether the site was found here, but I don't think a 404 is the right error. This is way down in a serializer util, we don't have enough context to know what should be returned to the user. This situation is most likely due to a programming error here. I think we should log an error that says something like "Tried to retrieve a site from context but no site found for the site_slug provided.", and then raise an APIException

Reply:
We should also log and raise if we get past the two conditions and haven't returned anything